### PR TITLE
Implement build_and_post_block in pythonic way

### LINKED ISSUE #167: Create pythonic build_and_post_block
## Description

Instead of using a dict full of options the args should be passed in a pythonic way.
Also we need an Input and Output class

Part of #58 

## Are you planning to do it yourself in a pull request?

Yes.

### PR DESCRIPTION
**Pull request by Dr-Electron**
# Description of change

I implemented the build_and_post_block function in a pythonic way.
This involved creating classes/enums for unlock conditions, features, native tokens and more.

## Links to any relevant issues

Fixes #167 

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

Tested with examples

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Please fix conflicts

**Review by kwek20 — DISMISSED** [review]

File: documentation/client/docs/libraries/python/references/iota_client/client.md [hunk 1]
```diff
@@ -82,7 +82,18 @@ Generate addresses.
 ### build\_and\_post\_block
 
 ```python
-def build_and_post_block(secret_manager=None, options=None)
+def build_and_post_block(secret_manager=None,
```
Comment (thibault-martinez) [inline documentation/client/docs/libraries/python/references/iota_client/client.md:85]: The actual `build_and_post_block` function is not changed in this PR ?

Comment (Dr-Electron) [reply documentation/client/docs/libraries/python/references/iota_client/client.md:85]: God damnit. That probably has been lost in one of the latest rebases. Sorry.
Let me check 😓

Comment (thibault-martinez) [reply documentation/client/docs/libraries/python/references/iota_client/client.md:85]: wen

Comment (Dr-Electron) [reply documentation/client/docs/libraries/python/references/iota_client/client.md:85]: Today. But i need to redo it as it seems to be lost in history 🥲

File: documentation/client/docs/libraries/python/how_to/_07_create_block/_key_api_ref_links.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
-### `Client.build_and_post_block(secret_manager=None, options=None)`
+### `Client.build_and_post_block(secret_manager=None)`
 
 You can create a simple empty block by calling
-the [`Client.build_and_post_block(secret_manager=None, options=None)`](./../libraries/python/api_reference#build_and_post_block)
+the [`Client.build_and_post_block(secret_manager=None)`](./../libraries/python/api_reference#build_and_post_block)
```
Comment (thibault-martinez) [inline documentation/client/docs/libraries/python/how_to/_07_create_block/_key_api_ref_links.md:4]: Should add the new params?

File: sdk/src/client/bindings/python/examples/08_data_block.py [hunk 3]
```diff
@@ -3,12 +3,7 @@
...
-    # `hello` hex encoded
-    "tag": '0x68656c6c6f',
-    "data": '0x68656c6c6f',
-}
-
-# Create and post a block with a tagged data payload
-block = client.build_and_post_block(None, options)
+# Create and post a block with a tagged data payload. Data is 'hello' hex encoded
+block = client.build_and_post_block(
+    None, tag='0x68656c6c6f', data='0x68656c6c6f')
```
Comment (thibault-martinez) [inline sdk/src/client/bindings/python/examples/08_data_block.py]: What's that first `None` ? Shouldn't it be named ?

File: documentation/client/docs/libraries/python/how_to/_07_create_block/_key_api_ref_links.md (hunk 2 above)
Comment (Dr-Electron) [reply documentation/client/docs/libraries/python/how_to/_07_create_block/_key_api_ref_links.md:4]: I actually still don't know how this page is generated 😆. I think there I just need to use pydoc or something? @Thoralf-M do you know?

File: sdk/src/client/bindings/python/examples/08_data_block.py (hunk 3 above)
Comment (Dr-Electron) [reply sdk/src/client/bindings/python/examples/08_data_block.py]: It's the secret manager which is in position 0. So it will work. But you are right, it may be harder to understand. Should I make it named? Wdyt?

File: documentation/client/docs/libraries/python/how_to/_07_create_block/_key_api_ref_links.md (hunk 2 above)
Comment (Thoralf-M) [reply documentation/client/docs/libraries/python/how_to/_07_create_block/_key_api_ref_links.md:4]: This page is not generated, it's manually written no?
But for the API ref it's here explained in the readme :trollface: 
https://github.com/iotaledger/iota-sdk/blob/develop/sdk/src/client/bindings/python/README.md#generate-api-references

File: sdk/src/client/bindings/python/examples/08_data_block.py (hunk 3 above)
Comment (thibault-martinez) [reply sdk/src/client/bindings/python/examples/08_data_block.py]: Yes, named

Comment (Dr-Electron) [reply sdk/src/client/bindings/python/examples/08_data_block.py]: Done

File: documentation/client/docs/libraries/python/how_to/_07_create_block/_key_api_ref_links.md (hunk 2 above)
Comment (Dr-Electron) [reply documentation/client/docs/libraries/python/how_to/_07_create_block/_key_api_ref_links.md:4]: I thought it was also autogenerated. But yeah will change manually then.
The API thingy is probably still completely broken anyway :trollface:

Comment (Dr-Electron) [reply documentation/client/docs/libraries/python/how_to/_07_create_block/_key_api_ref_links.md:4]: Ok sorry, took me some time. But I had to figure out if this docs are even used right now, because I couldn't find them.
TL;DR: They aren't 😆 .

They were just an idea back then to reference to the API documentation. But Rust wasn't published yet. 🤔 

But anyway as this would/will(?) link to the actual function doc it should be fine to keep it simple

Comment (Dr-Electron) [reply documentation/client/docs/libraries/python/how_to/_07_create_block/_key_api_ref_links.md:4]: Created an [issue](https://github.com/iotaledger/iota-sdk/issues/273) for those files btw.

File: sdk/src/client/bindings/python/iota_client/client.py [hunk 4]
```diff
@@ -857,3 +919,25 @@ def as_dict(self):
             config['maximum_supply'] = str(hex(config['maximum_supply']))
 
         return config
+
+class Output():
```
Comment (Thoralf-M) [inline sdk/src/client/bindings/python/iota_client/client.py]: I think we want to use this name for an actual output type later, so maybe call it `AddressWithAmount` as we do in the Rust lib

Comment (Dr-Electron) [reply sdk/src/client/bindings/python/iota_client/client.py]: Yeah makes sense. Thought about keeping it simple but knew about the danger of you wanting to use it in the future.
42 Gi and you can have the name :trollface:

Comment (Thoralf-M) [reply sdk/src/client/bindings/python/iota_client/client.py]: Greedy, 1.2 Gi or I will close this PR :trollface:

Comment (Dr-Electron) [reply sdk/src/client/bindings/python/iota_client/client.py]: Deal and done

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]
